### PR TITLE
Country item details page

### PR DESCRIPTION
**Pull request by predeyo**
Country item details page
Style bleed fixed from Country grid component
Css reset updated

### CODE REVIEW

**Comment by SumitKPandit** [comment]
The max-width of the container should match with that of the header on all screen sizes. Another thing I noticed was that the page works correctly only if we search for a country on the home page and then navigate by clicking on the link. If we directly type the same thing in URL bar, it doesn't work.

File: src/pages/CountryDetailsPage/CountryDetailsPage.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                            <li>
+                                <b>Currencies:</b>{' '}
+                                {country.currencies.map(
+                                    currency => currency.name
+                                )}
+                            </li>
+                            <li>
+                                <b>Languages:</b>{' '}
+                                {country.languages.map(
+                                    language => `${language.name} `
```
Comment (SumitKPandit) [inline src/pages/CountryDetailsPage/CountryDetailsPage.js:65]: Maybe we can use a comma between languages.

**Comment by predeyo** [comment]
> Isn't there a way to just pass info of only one country to the detail page, instead of having all data there and searching again?

The thing is we are using router, for example just navigate to a link on some country details page by typing the address, the first thing we have to do is to fetch the country, and we know only the route parameter and list of countries. 

Alternatively we could just fetch a single country from API, but App is already fetching all of them as Details page is the child component of App. In this case if we navigate to a details page from Country grid, we would be making extra API calls.

What I think could be the best thing to do here is to make use of Context API that could solve multiple problems, including we could get rid of prop drilling for "theme" state and other things, but I think that could be an other PR :)

I might be totally wrong so I'm open for alternative suggestions!

File: src/pages/CountryDetailsPage/CountryDetailsPage.js (hunk 1 above)
Comment (predeyo) [reply src/pages/CountryDetailsPage/CountryDetailsPage.js]: done

**Pull request merged**